examples/trusted_database/module_1/rust/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Trusted Database Handler module.
+//!
+//! In the current implementation clients send their location coordinates (latitude and longitude)
+//! and the Handler Node returns the location of the closest Point Of Interest.
+//!
+//! The Handler Node searches for the closest Point Of Interest in the database received from the
+//! Main Node.
+
+pub mod proto {
+    pub mod oak {
+        pub use oak::proto::oak::invocation;
+        pub mod examples {
+            pub mod trusted_database {
+                include!(concat!(
+                    env!("OUT_DIR"),
+                    "/oak.examples.trusted_database.rs"
+                ));
+                include!(concat!(
+                    env!("OUT_DIR"),
+                    "/oak.examples.trusted_database_command.rs"
+                ));
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests;
+
+use log::{debug, error};
+use oak::{grpc, io::ReceiverExt};
+use proto::oak::examples::trusted_database::{
+    ListPointsOfInterestRequest, ListPointsOfInterestResponse, Location, PointOfInterest,
+    TrustedDatabase, TrustedDatabaseCommand, TrustedDatabaseDispatcher,
+};
+
+// Error messages.
+const NO_LOCATION_ERROR: &str = "Location is not specified";
+const EMPTY_DATABASE_ERROR: &str = "Empty database";
+
+/// Oak Node that contains a copy of the database .
+pub struct TrustedDatabaseHandlerNode {
+    points_of_interest: Vec<PointOfInterest>,
+}
+
+/// A gRPC service implementation for the Private Information Retrieval example.
+impl TrustedDatabase for TrustedDatabaseHandlerNode {
+    /// Find the nearest Point Of Interest based on linear search in the database.
+    fn list_points_of_interest(
+        &mut self,
+        request: ListPointsOfInterestRequest,
+    ) -> grpc::Result<ListPointsOfInterestResponse> {
+        debug!("Received request: {:?}", request);
+        let location = request.location.ok_or_else(|| {
+            let err = grpc::build_status(grpc::Code::InvalidArgument, &NO_LOCATION_ERROR);
+            error!("{:?}", err);
+            err
+        })?;
+        let nearest_point = self.points_of_interest.iter().fold(
```

Comment:
Would it make sense to use https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min_by or https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min_by_key here instead of `fold`?

Reply:
I think you are already comparing pairs of elements anyways, so `min_by` seems fine? Anyway not a big deal.

Reply:
Added to `third_party/data/cycle-hire-availability/LICENSE`